src/concurrency/data_race.rs

```diff
@@ -981,6 +1020,8 @@ impl VClockAlloc {
         &self,
         alloc_id: AllocId,
         access_range: AllocRange,
+        read_type: NaReadType,
+        ty: Option<Ty<'_>>,
```

Comment:
Please explain the meaning of this parameter in the doc comment, and explicitly say that it is for diagnostics only.